drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -956,13 +958,16 @@ static void mmcsd_decode_scr(FAR struct mmcsd_state_s *priv, uint32_t scr[2])
    *   DATA_STATE_AFTER_ERASE 55:55 1-bit erase status
    *   SD_SECURITY            54:52 3-bit SD security support level
    *   SD_BUS_WIDTHS          51:48 4-bit bus width indicator
-   *   Reserved               47:32 16-bit SD reserved space
+   *   Reserved               47:34 14-bit SD reserved space
+   *   CMD_SUPPORT            33:32 2-bit command supported (bit33 for cmd23)
    */
 
 #ifdef CONFIG_ENDIAN_BIG  /* Card transfers SCR in big-endian order */
   priv->buswidth     = (scr[0] >> 16) & 15;
+  priv->cmd23support =  scr[0]        & 2;
```

Comment:
`cmd23support` is defined as 1 bit, but 2 bit result is assigned. Wouldn't that be an issue here? Shouldn't code be either `priv->cmd23support = (scr[0] >> 1) & 1;` or `priv->cmd23support = (scr[0] & 2) != 0 ? 1 : 0;`?

Reply:
Maybe some implicit conversion to `bool` is applied.

Reply:
It looks like overflow to me. The result will always be zero.
Both your suggestions should fix it.

Reply:
Yes. Fix is already applied